src/libstd/sys/redox/fs.rs

```diff
@@ -447,7 +447,10 @@ pub fn stat(p: &Path) -> io::Result<FileAttr> {
 }
 
 pub fn lstat(p: &Path) -> io::Result<FileAttr> {
-    stat(p)
+    let fd = cvt(syscall::open(p.to_str().unwrap(),
+                               syscall::O_CLOEXEC | syscall::O_STAT | syscall::O_NOFOLLOW))?;
```

Comment:
FWIW this should probably also specify `O_RDONLY` to [obey posix](http://pubs.opengroup.org/onlinepubs/009695399/functions/open.html) (since redox seems to be posixish).

(it's an interesting dilemma whether redox should reject that syscall as-is - it's technically invalid, but linux would accept it because `O_RDONLY` is defined as 0, so is always implicit. In redox it's [defined](https://github.com/redox-os/syscall/blob/f3c97bee78f2a56c4e31dc0e2272d0e5daa1180c/src/flag.rs#L33) as 1, so there's a choice between correctness and Linux program compatibility)